src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantAdvertService.java

```diff
@@ -31,18 +32,24 @@ protected static String getExternalSubmissionUrl(GrantAdvert advert) {
 
     public GetGrandAdvertDto getAdvertByContentfulSlug(String contentfulSlug) {
 
-        final GrantAdvert advert = grantAdvertRepository.findByContentfulSlug(contentfulSlug)
-                .orElseThrow(() -> new NotFoundException("Advert with slug " + contentfulSlug + " not found"));
+        final Optional<GrantAdvert> advert = grantAdvertRepository.findByContentfulSlug(contentfulSlug);
+        if(advert.isEmpty()){
```

Comment:
This logic could very easily lead to other problems because it's not actually confirming if the grant exists in Contentful, it's just guessing. I am wondering if we could pull in the Contentful dependency and do this check properly? If you are too busy then I don't mind doing this work.

Reply:
The slug we pass to this method comes from a call to contentful done in find a grant and passed to apply through query param, so shouldn't be a problem for this scenario.
But I see your point in case we would need to use this method for other scenarios, so yeah better add a check also in apply

Reply:
Added code to check contentful for whether or not a grant exists